crates/interpreter/src/instructions/bitwise.rs

```diff
@@ -125,26 +117,21 @@ pub fn sar<SPEC: Spec>(interpreter: &mut Interpreter, _host: &mut dyn Host) {
     gas!(interpreter, gas::VERYLOW);
     pop_top!(interpreter, op1, op2);
 
-    let value_sign = i256_sign::<true>(op2);
+    let value_sign = i256_sign_compl(op2);
 
     *op2 = if *op2 == U256::ZERO || op1 >= U256::from(256) {
         match value_sign {
             // value is 0 or >=1, pushing 0
             Sign::Plus | Sign::Zero => U256::ZERO,
             // value is <0, pushing -1
-            Sign::Minus => two_compl(U256::from(1)),
+            Sign::Minus => U256::MAX,
         }
     } else {
+        const ONE: U256 = U256::from_limbs([1, 0, 0, 0]);
         let shift = usize::try_from(op1).unwrap();
-
         match value_sign {
-            Sign::Plus | Sign::Zero => *op2 >> shift,
-            Sign::Minus => {
-                let shifted = ((op2.overflowing_sub(U256::from(1)).0) >> shift)
-                    .overflowing_add(U256::from(1))
-                    .0;
-                two_compl(shifted)
-            }
+            Sign::Plus | Sign::Zero => op2.wrapping_shr(shift),
+            Sign::Minus => two_compl(op2.wrapping_sub(ONE).wrapping_shr(shift).wrapping_add(ONE)),
```

Comment:
Seems fine if we are OK with going from overflowing -> wrapping

Reply:
We can simplify this to `((op2-ONE)>>shift)+1)` as checks for `op2 == zero` and `op1 >= 256` are done and `shr` in background calls `overflowing_shr` inside ruint.

Reply:
Makes sense in the context that ruint operations are overloaded to be wrapping, but it might not be obvious without this knowledge. I wrote them with explicit `wrapping`, normally I'd use operators but this is one of the very few instances where I'd rather use explicit methods

Reply:
`wrapper_*` are nicer here then `overflowing_*` so I will just merge PR.
But I am curious why you think this is one of those instances, bearing in mind that we already check for `op2 == zero and op1 >= 256`